File: day-09/part-1/badouralix.py
from runners.python import Submission

class BadouralixSubmission(Submission):

	def run(self, s):
		# :param s: input in string format
		# :return: solution flag
		current_value = 0
		result = 0
		garbage = False
		skip = False

		for char in s:
			if skip:
				skip = False
			else:
				if char == '!':
					skip = True
				elif char == '>':
					garbage = False
				elif not garbage:
					if char == '{':
						current_value += 1
					elif char == '<':
						garbage = True
					elif char == '}':
						result += current_value
						current_value -= 1
						
		return result

File: day-09/part-1/badouralix.c
#include <stdio.h>
#include <stdbool.h>

const int run(char* s)
{
    int current_value = 0,
        result = 0;
    bool garbage = false;

    for (char* c = --s; *++c != '\0';) {
        switch (*c) {
        case '!':
            c++;
            break;
        case '{':
            if (!garbage) current_value++;
            break;
        case '}':
            if (!garbage) result += current_value--;
            break;
        case '<':
            garbage = true;
            break;
        case '>':
            garbage = false;
            break;
        default:
            break;
        }
    }

    return result;
}

int main(int argc, char** argv)
{
    printf("%d\n", run(argv[1]));
    return 0;
}
